numpy/core/tests/test_umath.py

```diff
@@ -935,6 +935,12 @@ def test_sign(self):
             assert_equal(res, tgt)
             assert_equal(out, tgt)
 
+    def test_sign_dtype_object(self):
```

Comment:
We typically add a comment back to the relevant issue, e.g. `# gh-6229` in this case.